x/gov/handler_test.go

```diff
@@ -1020,24 +998,59 @@ func TestTally(t *testing.T) {
 			},
 			ExpResult: Proposal_Accepted,
 			PostChecks: func(t *testing.T, db weave.KVStore) {
-				elect, err := NewElectorateBucket().GetElectorate(db, weavetest.SequenceID(1))
+				_, obj, err := NewElectorateBucket().GetLatestVersion(db, weavetest.SequenceID(1))
 				if err != nil {
 					t.Fatalf("unexpected error: %+v", err)
 				}
+				elect, _ := asElectorate(obj)
+				if exp, got := uint32(3), elect.Version; exp != got {
+					t.Errorf("expected %v but got %v", exp, got)
+				}
 				got := elect.Electors
-				exp := []Elector{{alice, 10}, {bobby, 10}}
+				exp := []Elector{{alice, 10}, {bobby, 10}, {charlie, 2}}
 				sortByAddress(exp)
 				if !reflect.DeepEqual(exp, got) {
 					t.Errorf("expected %v but got %v", exp, got)
 				}
-				if exp, got := uint64(20), elect.TotalElectorateWeight; exp != got {
+				if exp, got := uint64(22), elect.TotalElectorateWeight; exp != got {
 					t.Errorf("expected %v but got %v", exp, got)
 				}
 			},
 		},
+		"Does not complete tally when executor fails": {
+			Init: func(t *testing.T, db weave.KVStore) {
+				// update electorate for a new version without Alice
+				bucket := NewElectorateBucket()
+				_, obj, err := bucket.GetLatestVersion(db, weavetest.SequenceID(1))
+				if err != nil {
+					t.Fatalf("unexpected error: %+v", err)
+				}
+				e, _ := asElectorate(obj)
+				e.Electors = []Elector{{bobby, 10}}
+				e.TotalElectorateWeight = 10
+
+				if _, err := bucket.Update(db, weavetest.SequenceID(1), e); err != nil {
+					t.Fatalf("unexpected error: %+v", err)
+				}
+			},
+			Mods: func(ctx weave.Context, p *Proposal) {
```

Comment:
I think you also need `WithProposal: true`, so the proposal is created and Mods executed

Reply:
That is used in `TestUpdateElectorate`. 😅